hoodie-client/src/main/java/com/uber/hoodie/table/HoodieTable.java

```diff
@@ -190,49 +192,10 @@ public HoodieActiveTimeline getActiveTimeline() {
   }
 
   /**
-   * Get the commit timeline visible for this table
+   * Return the index
    */
-  public HoodieTimeline getCommitsTimeline() {
```

Comment:
whats the motivation for moving these methods alone? We need to define a principle for this move, so that the classes can stay in tact. I am little concerned over time, we will keep adding new helpers on both classes - HoodieTable, HoodieTableMetaClient - and it will become unclear what to use when

Reply:
Moved the timeline to metaclient since many HoodieIndex internal implementations require the timeline. Since index is moved inside table, instantiating the table again inside the index to gain access to the timeline seems counter-intuitive, so moved the these to metaclient. Also, did not do this blindly. I think as we move towards making changes such that we have a timeline server, it seems like the timeline is more of metadata which can belong in metaclient, any such hoodie metadata operations henceforth could go into metaclient. WDYT ?
@vinothchandar

Reply:
There are two things inside HoodieTable : methods that query the timeline (these could move to TableMetaClient as you mention), FileSystemView (which is the more tricky one). Moving that into TableMetaClient does not make sense to me and thats the issue I am bringing up.

Reply:
Not following :( The FileSystemView hasn't been moved to HoodieTableMetaClient. Where do you see that ? 
I think what you're saying is the use of new HoodieTableFileSystemView in HoodieBloomIndex. The reason for that is that passing in the hoodieTable to HoodieBloomIndex seemed counter-intuitive since index is part of HoodieTable. So now since we cannot get the ROView due to absence of HoodieTable, I create a new instance. (the getROView() method call from hoodie table also gets a new instance and so there is no concern around caching that). 
But now that you speak of this, to reduce the listStatus call, it might be better to pass the hoodietable to the index. Does that make sense ?

@vinothchandar

Reply:
>>to reduce the listStatus call, it might be better to pass the hoodietable to the index. Does that make sense ?
This was my point all along.. We should be able to pass in HoodieTable or the FileSystemView into the index constructor or something.. 

On objects referring to each other, as long as you don't instantiate them recursively in both constructors, it should be okay? i,e init HoodieIndex from HoodieTable constructor, passing in `this`. I think better design would be simply pass in what's actually needed i.e TableFileSystemView. 

Is that any clearer?